core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java

```diff
@@ -38,22 +38,37 @@
     @Override
 
     public BidirectionalActions<A> apply(Problem<A, S> originalProblem, Problem<A, S> reverseProblem) {
-        Node<A, S> node = newRootNode(originalProblem.initialState(), 0);
+        Node<A, S> node = newRootNode(originalProblem.initialState(), 0); //node -> starting point 
+        //if starting node is the goal node then return bidirectionalAction 
```

Comment:
please avoid obsolete comments http://google.com/#q=clean+code+comments
use comments where the code isn't self-explanatory